tests/test_pw_hash.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2013 Donald Stufft and individual contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+
+import pytest
+
+import nacl.encoding
+import nacl.pw_hash
+
+from nacl.bindings.crypto_pwhash_scryptsalsa208sha256 import (
+    crypto_pwhash_scryptsalsa208sha256,
+    crypto_pwhash_scryptsalsa208sha256_str_verify,
+)
+
+
+@pytest.mark.parametrize(("size", "password", "salt",
+                          "opslimit", "memlimit",
+                          "expected"), [
+        (
+            32,
+            b"The quick brown fox jumps over the lazy dog.",
+            b"ef537f25c895bfa782526529a9b63d97",
+            20000,
+            (2 ** 20) * 100,
+            (b"\x10e>\xc8A8\x11\xde\x07\xf1\x0f\x98"
+             b"EG\xe6}V]\xd4yN\xae\xd3P\x87yP\x1b\xc7+n*")
```

Comment:
Inherited from Malthe Borch's d889f85 .

Just double checked with cpython3.6 hashlib implementation, by manually running libsodium's 
   pickparams(...) and finding out that with opslimit=20000 and memlimit=(2 ** 20) * 100 
n = 128, r = 8, p =1
```
>>> hashlib.scrypt(b'The quick brown fox jumps over the lazy dog.',
                               salt=b"ef537f25c895bfa782526529a9b63d97",
                               n=1024, r=8, p=1, maxmem = (2**20)*100, dklen=32)
b'\x10e>\xc8A8\x11\xde\x07\xf1\x0f\x98EG\xe6}V]\xd4yN\xae\xd3P\x87yP\x1b\xc7+n*'
>>>
```